src/Spice86.Core/Emulator/InterruptHandlers/Dos/DosInt21Handler.cs

```diff
@@ -517,18 +530,35 @@ public class DosInt21Handler : InterruptHandler {
         return _cp850CharSet.GetString(characterBytes.ToArray());
     }
     
+    /// <summary>
+    /// Gets an ASCIZ pathname containing the current DOS directory in the address at DS:DI. <br/>
+    /// Params: <br/>
+    /// DL = drive number (0x0: default, 0x1: A:, 0x2: B:, 0x3: C:, ...)
+    /// </summary>
+    /// <remarks>
+    /// Does not include a drive, or the initial backslash
+    /// </remarks>
+    /// <returns>
+    /// DS:DI = ASCIZ pathname containing the current DOS directory. <br/>
+    /// CF is clear if sucessful. <br/>
+    /// CF is set on error. Possible error code in AX: 0xF (InvalidDrive).
+    /// </returns>
+    /// <param name="calledFromVm">Whether the method was called by the emulator.</param>
     public void GetCurrentDirectory(bool calledFromVm) {
-        SetCarryFlag(false, calledFromVm);
+        uint responseAddress = MemoryUtils.ToPhysicalAddress(_state.DS, _state.SI);
+        DosFileOperationResult result = _dosFileManager.GetCurrentDir(_state.DL, out string currentDir);
         if (_loggerService.IsEnabled(LogEventLevel.Verbose)) {
-            _loggerService.Verbose("GET CURRENT DIRECTORY {ResponseAddress}",
-                ConvertUtils.ToSegmentedAddressRepresentation(_state.DS, _state.SI));
+            _loggerService.Verbose("GET CURRENT DIRECTORY {ResponseAddress}: {CurrentDpsDirectory}",
+                ConvertUtils.ToSegmentedAddressRepresentation(_state.DS, _state.SI), currentDir);
         }
-        uint responseAddress = MemoryUtils.ToPhysicalAddress(_state.DS, _state.SI);
-        // Fake that we are always at the root of the drive (empty String)
-        _memory.UInt8[responseAddress] = 0;
+        _memory.LoadData(responseAddress, Encoding.ASCII.GetBytes(currentDir).Concat(Encoding.ASCII.GetBytes("0")).ToArray());
```

Comment:
Why not using Memory.SetZeroTerminatedString?

Reply:
Done.